framework/web/View.php

```diff
@@ -470,6 +456,57 @@ public function registerJs($js, $position = self::POS_READY, $key = null)
         }
     }
 
+    /**
+     * Registers a JS or CSS file.
+     *
+     *
+     * @param string $url the JS file to be registered.
+     * @param string $type type (js or css) of the file.
+     * @param array $options the HTML attributes for the script tag. The following options are specially handled
+     * and are not treated as HTML attributes:
+     * @param string $key the key that identifies the JS script file. If null, it will use
+     * $url as the key. If two JS files are registered with the same key at the same position, the latter
+     * will overwrite the former. Note that position option takes precedence, thus files registered with the same key,
+     * but different position option will not override each other.
+     * @throws InvalidConfigException
+     */
+    private function registerFile($type, $url, $options = [], $key = null)
+    {
+        $url = Yii::getAlias($url);
+        $key = $key ?: $url;
+        $depends = ArrayHelper::remove($options, 'depends', []);
+        $position = ArrayHelper::remove($options, 'position', self::POS_END);
+        try { // avoid The directory does not exist if basePath is not set properly. Do we really need it?
+            $managerAppendTimestamp = $this->getAssetManager()->appendTimestamp;
+        } catch (InvalidConfigException $e) {
+            $depends = null; // the AssetManager is not available
+            $managerAppendTimestamp = false;
+        }
+        $appendTimestamp = ArrayHelper::remove($options, 'appendTimestamp', $managerAppendTimestamp);
+
+        if (empty($depends)) {
+            // register directly without AssetManager
+            if ($appendTimestamp && ($timestamp = @filemtime(Yii::getAlias('@webroot/' . ltrim($url, '/'), false))) > 0) {
```

Comment:
URL could be absolute, couldn't it?

Reply:
Yes, it could. I had checked it in https://github.com/yiisoft/yii2/blob/767ccd1d9541c8d2a35f78aa4dece39cb821da00/tests/framework/web/ViewTest.php#L336
We just won't find the file on @webroot/https... and it will be without timestamp

Reply:
I think it's better to check with `Url::isRelative()` and not try to call `filemtime()`.